src/Functions/readWkb.cpp

```diff
@@ -0,0 +1,253 @@
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeCustomGeo.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+
+#include "Common/Exception.h"
+#include <Common/WKB.h>
+#include "Columns/ColumnFixedString.h"
+#include "Columns/IColumn.h"
+#include "DataTypes/DataTypeFixedString.h"
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <variant>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+}
+
+namespace
+{
+
+template <class DataTypeName, class Geometry, class Serializer, class NameHolder>
+class FunctionReadWKB : public IFunction
+{
+public:
+    explicit FunctionReadWKB() = default;
+
+    static constexpr const char * name = NameHolder::name;
+
+    String getName() const override { return "readWKB"; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (checkAndGetDataType<DataTypeString>(arguments[0].get()) == nullptr
+            && checkAndGetDataType<DataTypeFixedString>(arguments[0].get()) == nullptr)
+        {
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "First argument should be String");
+        }
+
+        return DataTypeFactory::instance().get(DataTypeName().getName());
+    }
+
+    ColumnPtr
+    executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & /*result_type*/, size_t input_rows_count) const override
+    {
+        if (checkColumn<ColumnString>(*arguments[0].column))
+        {
+            const auto & column_string = checkAndGetColumn<ColumnString>(*arguments[0].column);
+            return executeColumnImpl(column_string, input_rows_count);
+        }
+        else
+        {
+            const auto & column_string = checkAndGetColumn<ColumnFixedString>(*arguments[0].column);
+            return executeColumnImpl(column_string, input_rows_count);
+        }
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionReadWKB<DataTypeName, Geometry, Serializer, NameHolder>>(); }
+
+private:
+    template <typename T>
+    ColumnPtr executeColumnImpl(const T & column, size_t input_rows_count) const
+    {
+        Serializer serializer;
+        for (size_t i = 0; i < input_rows_count; ++i)
+        {
+            auto str = column.getDataAt(i).toString();
```

Comment:
You don't need to copy the data here. You can use `getDataAt(i)` and leave it as a StringRef. Then use s.data, and s.size